src/software/world/field.h

```diff
@@ -21,36 +21,36 @@ class Field
     /**
      * Constructs a new field with the given dimensions
      *
-     * @param field_length the length of the playing area (along the x-axis)
-     * @param field_width the width of the playing area (along the y-axis)
-     * @param defense_length the length of the defense area (along the x-axis)
-     * @param defense_width the width of the defense area (along the y-axis)
-     * @param goal_width the width of the goal (along the y-axis)
-     * @param boundary_width the width/size of the boundary area between the edge of the
+     * @param field_x_length the length of the playing area (along the x-axis)
+     * @param field_y_length the length of the playing area (along the y-axis)
+     * @param defense_x_length the length of the defense area (along the x-axis)
+     * @param defense_y_length the length of the defense area (along the y-axis)
+     * @param goal_y_length the length of the goal (along the y-axis)
+     * @param boundary_y_length the length/size of the boundary area between the edge of the
      * playing area and the physical border/perimeter of the field
      * @param center_circle_radius the radius of the center circle
      * @param timestamp the Timestamp associated with the creation of the Field object
      */
-    explicit Field(double field_length, double field_width, double defense_length,
-                   double defense_width, double goal_width, double boundary_width,
+    explicit Field(double field_x_length, double field_y_length, double defense_x_length,
+                   double defense_y_length, double goal_y_length, double boundary_y_length,
                    double center_circle_radius, const Timestamp &timestamp,
                    unsigned int buffer_size = 20);
 
     /**
      * Updates the dimensions of the field. All units should be in metres.
      *
-     * @param field_length the length of the playing area (along the x-axis)
-     * @param field_width the width of the playing area (along the y-axis)
-     * @param defense_length the length of the defense area (along the x-axis)
-     * @param defense_width the width of the defense area (along the y-axis)
-     * @param goal_width the width of the goal (along the y-axis)
-     * @param boundary_width the width/size of the boundary area between the edge of the
+     * @param field_x_length the length of the playing area (along the x-axis)
+     * @param field_y_length the length of the playing area (along the y-axis)
+     * @param defense_x_length the length of the defense area (along the x-axis)
+     * @param defense_y_length the length of the defense area (along the y-axis)
+     * @param goal_y_length the length of the goal (along the y-axis)
+     * @param boundary_y_length the length/size of the boundary area between the edge of the
```

Comment:
I think the `boundary` parameter can stay roughly as is. It's the "buffer size" all the way around the field so isn't associated with x or y in particular. Maybe rename it to `boundary_buffer_distance`, 'boundary_buffer_size` or something like that? @garethellis0 thoughts on naming?

This will need to be changed everywhere in the PR

Reply:
Agreed, `boundary_buffer_distance` seems like the right terminology here.